terraform/modules/happy-stack-helm-eks/variables.tf

```diff
@@ -0,0 +1,285 @@
+      v.image_pull_policy == "Always" ||
+      v.image_pull_policy == "Never"
+    )])
+    error_message = "The image_pull_policy argument needs to be 'IfNotPresent', 'Always', or 'Never'."
+  }
+
+  validation {
+    condition = alltrue([for k, v in var.services : (
+      v.service_scheme == "HTTP" ||
+      v.service_scheme == "HTTPS"
+    )])
+    error_message = "The service_scheme argument needs to be 'HTTP' or 'HTTPS'."
+  }
+
+  validation {
+    condition = alltrue([for k, v in var.services : (
+      v.service_type == "EXTERNAL" ||
+      v.service_type == "INTERNAL" ||
+      v.service_type == "PRIVATE" ||
+      v.service_type == "IMAGE_TEMPLATE" ||
+      v.service_type == "TARGET_GROUP_ONLY" ||
+      v.service_type == "VPC"
+    )])
+    error_message = "The service_type argument needs to be 'EXTERNAL', 'INTERNAL', 'PRIVATE', 'VPC', or 'IMAGE_TEMPLATE'."
+  }
+  validation {
+    condition     = alltrue([for k, v in var.services : v.alb != null if v.service_type == "TARGET_GROUP_ONLY"])
+    error_message = "The service_type 'TARGET_GROUP_ONLY' requires an alb"
+  }
+  validation {
+    # The health check prefix needs to contain the service path for CONTEXT services, but not TARGET_GROUP_ONLY services.
+    condition     = alltrue([for k, v in var.services : startswith(v.health_check_path, trimsuffix(v.path, "*")) if v.service_type != "TARGET_GROUP_ONLY"])
+    error_message = "The health_check_path should start with the same prefix as the path argument."
+  }
+  validation {
+    condition     = alltrue(flatten([for k, v in var.services : [for path in flatten([for x, y in v.bypasses : y.paths]) : startswith(path, trimsuffix(v.path, "*"))]]))
+    error_message = "The bypasses.paths should all start with the same prefix as the path argument."
+  }
+  validation {
+    condition     = alltrue([for service in var.services : alltrue([for sidecar in service.sidecars : contains(["IfNotPresent", "Always", "Never"], sidecar.image_pull_policy)])])
+    error_message = "Value of a sidecar image_pull_policy needs to be 'IfNotPresent', 'Always', or 'Never'."
+  }
+  validation {
+    condition     = alltrue([for service in var.services : alltrue([for sidecar in service.sidecars : length(sidecar.health_check_path) > 0])])
+    error_message = "Value of a sidecar health_check_path must be a non-empty string."
+  }
+  validation {
+    condition     = alltrue([for service in var.services : alltrue([for sidecar in service.sidecars : sidecar.initial_delay_seconds > 0])])
+    error_message = "Value of a sidecar initial_delay_seconds must be a positive number."
+  }
+  validation {
+    condition     = alltrue([for service in var.services : alltrue([for sidecar in service.sidecars : sidecar.period_seconds > 0])])
+    error_message = "Value of a sidecar period_seconds must be a positive number."
+  }
+}
+
+variable "tasks" {
+  type = map(object({
+    image : string,
+    memory : optional(string, "10Mi"),
+    cpu : optional(string, "10m"),
```

Comment:
Is this a good default? This is the default in `happy-stack-eks` too but it seems kinda useless since pretty much nothing can run with these constraints. If we used a higher default it might save users from having to always provide an override

Reply:
Definitely not !

Reply:
As in not a good set of defaults!

Reply:
I've updated the defaults, but it is up to the application owner to adjust them to reflect real application needs.